File: Baekjoon/스택/10828 - 스택.cpp
#include<iostream>
#include<stack>

int main()
{
    int n, inputInt;
    std::string input;
    std::stack<int> s;

    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        std::cin >> input;
        if (input == "push")
        {
            scanf("%d", &inputInt);
            s.push(inputInt);
        }
        else if (input == "pop")
            if (s.empty())    printf("-1\n");
            else
            {
                printf("%d\n", s.top());
                s.pop();
            }
        else if (input == "size")
        {
            printf("%d\n", s.size());
        }
        else if (input == "empty")
            if (s.empty())    printf("1\n");
            else             printf("0\n");
        else if (input == "top")
            if (s.empty())   printf("-1\n");
            else    printf("%d\n", s.top());
    }
    system("pause");
}

File: Baekjoon/10828 - 스택.java
import java.util.Stack;
import java.util.StringTokenizer;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws Exception{

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		
		StringTokenizer st;
		
		int n = Integer.parseInt(br.readLine());
		
		Stack<Integer> stack = new Stack<>();
		
		while(n-- > 0)
		{
			st = new StringTokenizer(br.readLine(), " ");
			
			switch(st.nextToken()) {
			
			case "push":
				stack.add(Integer.parseInt(st.nextToken()));
				break;
			case "pop":
				if(!stack.isEmpty()) {
					sb.append(stack.peek()).append('\n');
					stack.pop();
				} else
					sb.append(-1).append('\n');
				break;
			case "size":
				sb.append(stack.size()).append('\n');
				break;
			case "empty":
				sb.append(stack.empty()==true?1:0).append('\n');
				break;
			case "top":
				if(!stack.isEmpty()) 
					sb.append(stack.peek()).append('\n');
				else
					sb.append(-1).append('\n');
				break;
			}
		}
		System.out.println(sb);
	}
}